CalPatRec/src/AgnosticHelixFinder_module.cc

```diff
@@ -729,47 +750,38 @@ namespace mu2e {
   //-----------------------------------------------------------------------------
   void AgnosticHelixFinder::tcHitsFill(size_t tc) {
 
-    size_t sortStartIndex = 0;
-
     cHit hit;
-    hit.circleError2 = 1.0;
-    hit.helixPhi = 0.0;
-    hit.helixPhiError2 = 0.0;
-    hit.helixPhiCorrection = 0;
-    hit.inHelix = false;
-    hit.used = false;
-    hit.isolated = false;
-    hit.averagedOut = false;
-    hit.notOnLine = true;
-    hit.uselessTripletSeed = false;
-    hit.notOnSegment = true;
 
     // push back stopping target if it is to be used
     if (_useStoppingTarget == true) {
       hit.hitIndice = HitType::STOPPINGTARGET;
+      hit.z = _stopTargPos.z();
       _tcHits.push_back(hit);
-      sortStartIndex++;
     }
 
     // push back calo cluster if it exists in time cluster
     if (_tcColl->at(tc).hasCaloCluster()) {
       hit.hitIndice = HitType::CALOCLUSTER;
-      _tcHits.push_back(hit);
       const art::Ptr<CaloCluster> cl = _tcColl->at(tc).caloCluster();
-      CLHEP::Hep3Vector gpos = _calorimeter->geomUtil().diskToMu2e(cl->diskID(), cl->cog3Vector());
-      CLHEP::Hep3Vector tpos = _calorimeter->geomUtil().mu2eToTracker(gpos);
-      double offset = _calorimeter->caloInfo().getDouble("diskCaseZLength");
-      offset += _calorimeter->caloInfo().getDouble("BPPipeZOffset");
-      offset += _calorimeter->caloInfo().getDouble("BPHoleZLength");
-      offset += _calorimeter->caloInfo().getDouble("FEEZLength");
-      offset /= 2.0;
-      _caloPos.SetCoordinates(tpos.x(), tpos.y(), tpos.z() - offset);
-      sortStartIndex++;
+      const CaloCluster*cluster = cl.get();
+      if (cluster != nullptr){
+        CLHEP::Hep3Vector gpos = _calorimeter->geomUtil().diskToMu2e(cl->diskID(), cl->cog3Vector());
+        CLHEP::Hep3Vector tpos = _calorimeter->geomUtil().mu2eToTracker(gpos);
+        double offset = _calorimeter->caloInfo().getDouble("diskCaseZLength");
```

Comment:
The calculation of offset is invariant.  It could be done at beginRun time and cached.  It also involves string comparisons so it will be slow.